File: testbed_ruby/test/associations/many_to_many_test.rb
require 'test/unit'
require './src-gen-umple/student_n'
require './src-gen-umple/mentor_n'

module CruiseTest
class ManyToManyTest < Test::Unit::TestCase
    
  def test_CreateNew
    s = StudentN.new(99)
    assert_equal(0,s.number_of_mentors)
    
    m = MentorN.new("blah")
    assert_equal(0,m.number_of_students)
  end

  def test_add_student
    m = MentorN.new("blah")
    s = StudentN.new(99)

    assert_equal true, m.add_student(s)
    
    assert_equal(m, s.get_mentor(0))
    assert_equal(s, m.get_student(0))
  end
  
  def test_add_studentToMultipleMentors
    m = MentorN.new("blah")
    m2 = MentorN.new("blah2")
    
    
    s = StudentN.new(99)
    s2 = StudentN.new(992)
    m.add_student(s)
    m2.add_student(s2)
    m2.add_student(s)
    s2.add_mentor(m)
    
    assert_equal(m, s.get_mentor(0))
    assert_equal(m2, s.get_mentor(1))
    assert_equal(m2, s2.get_mentor(0))
    assert_equal(m, s2.get_mentor(1))
  
    
    assert_equal(s, m.get_student(0))
    assert_equal(s2, m.get_student(1))
    assert_equal(s2, m2.get_student(0))
    assert_equal(s, m2.get_student(1))
  end
  
  # def test_addToNewMentor
  #   m = MentorN.new("blah")
  #   m2 = MentorN.new("blah2")
  #   s = StudentN.new(99)
  #   
  #   m.add_student(s)
  #   m2.add_student(s)
  #   
  #   assert_equal(2, s.number_of_mentors)
  #   assert_equal(m, s.mentor(0))
  #   assert_equal(m2, s.mentor(1))
  # 
  #   assert_equal(1, m.number_of_students)
  #   assert_equal(1, m2.number_of_students)
  # 
  #   assert_equal(s, m.student(0))
  #   assert_equal(s, m2.student(0))
  # end  
  # 
  # 
  # def test_doNotRemoveFromExistingMentor
  #   m = MentorN.new("blah")
  #   m2 = MentorN.new("blah2")
  # 
  #   s1 = StudentN.new(99)
  #   s2 = StudentN.new(98)
  #   s3 = StudentN.new(97)
  #   
  #   m.add_student(s1)
  #   m.add_student(s2)
  #   m.add_student(s3)
  # 
  #   assert_equal(3, m.number_of_students)
  #   assert_equal(0, m2.number_of_students)
  #   
  #   m2.add_student(s1)
  # 
  #   assert_equal(3, m.number_of_students)
  #   assert_equal(s1, m.student(0))
  #   assert_equal(s2, m.student(1))
  #   assert_equal(s3, m.student(2))
  #   
  #   assert_equal(1, m2.number_of_students)
  #   assert_equal(s1, m2.student(0))
  #   
  #   assert_equal(m, s1.mentor(0))
  #   assert_equal(m2, s1.mentor(1))
  #   
  #   assert_equal(m, s2.mentor(0))
  #   assert_equal(m, s3.mentor(0))
  # end   
  # 
  # 
  # def test_remove_student
  #   m = MentorN.new("blah")
  #   s = StudentN.new(99)
  #   
  #   m.add_student(s)
  #   m.remove_student(s)
  #   
  #   assert_equal(0,s.number_of_mentors)
  #   assert_equal(0,m.number_of_students)
  # end  
  # 
  # 
  # def test_deleteAll
  #   m = MentorN.new("blah")
  #   m2 = MentorN.new("blah2")
  #   
  #   s = StudentN.new(99)
  #   s2 = StudentN.new(98)
  #   s3 = StudentN.new(97)
  #   
  #   m.add_student(s)
  #   m.add_student(s2)
  #   m.add_student(s3)
  # 
  #   m2.add_student(s2)
  #   
  #   m.delete
  # 
  #   assert_equal(0,m.number_of_students)
  #   assert_equal(1,m2.number_of_students)
  #   
  #   assert_equal(0,s.number_of_mentors)
  #   assert_equal(1,s2.number_of_mentors)
  #   assert_equal(0,s3.number_of_mentors)
  # end  
  
end
end

File: testbed_php/test/associations/ManyToManyTest.php
<?php

class ManyToManyTest extends UnitTestCase
{
    
  function test_CreateNew()
  {
    $s = new StudentN(99);
    $this->assertEqual(0,$s->numberOfMentors());
    
    $m = new MentorN("blah");
    $this->assertEqual(0,$m->numberOfStudents());
  }
  
  function test_addStudent()
  {
    $m = new MentorN("blah");
    $s = new StudentN(99);
    $m->addStudent($s);
    
    $this->assertEqual($m, $s->getMentor_index(0));
    $this->assertEqual($s, $m->getStudent_index(0));
  }

  function test_addStudentToMultipleMentors()
  {
    $m = new MentorN("blah");
    $m2 = new MentorN("blah2");
    
    
    $s = new StudentN(99);
    $s2 = new StudentN(992);
    $m->addStudent($s);
    $m2->addStudent($s2);
    $m2->addStudent($s);
    $s2->addMentor($m);
    
    $this->assertEqual($m, $s->getMentor_index(0));
    $this->assertEqual($m2, $s->getMentor_index(1));
    $this->assertEqual($m2, $s2->getMentor_index(0));
    $this->assertEqual($m, $s2->getMentor_index(1));

    
    $this->assertEqual($s, $m->getStudent_index(0));
    $this->assertEqual($s2, $m->getStudent_index(1));
    $this->assertEqual($s2, $m2->getStudent_index(0));
    $this->assertEqual($s, $m2->getStudent_index(1));
  }
  

  function test_addToNewMentor()
  {
    $m = new MentorN("blah");
    $m2 = new MentorN("blah2");
    $s = new StudentN(99);
    
    $m->addStudent($s);
    $m2->addStudent($s);
    
    $this->assertEqual(2, $s->numberOfMentors());
    $this->assertEqual($m, $s->getMentor_index(0));
    $this->assertEqual($m2, $s->getMentor_index(1));

    $this->assertEqual(1, $m->numberOfStudents());
    $this->assertEqual(1, $m2->numberOfStudents());

    $this->assertEqual($s, $m->getStudent_index(0));
    $this->assertEqual($s, $m2->getStudent_index(0));
  }  
  

  function test_doNotRemoveFromExistingMentor()
  {
    $m = new MentorN("blah");
    $m2 = new MentorN("blah2");

    $s1 = new StudentN(99);
    $s2 = new StudentN(98);
    $s3 = new StudentN(97);
    
    $m->addStudent($s1);
    $m->addStudent($s2);
    $m->addStudent($s3);

    $this->assertEqual(3, $m->numberOfStudents());
    $this->assertEqual(0, $m2->numberOfStudents());
    
    $m2->addStudent($s1);

    $this->assertEqual(3, $m->numberOfStudents());
    $this->assertEqual($s1, $m->getStudent_index(0));
    $this->assertEqual($s2, $m->getStudent_index(1));
    $this->assertEqual($s3, $m->getStudent_index(2));
    
    $this->assertEqual(1, $m2->numberOfStudents());
    $this->assertEqual($s1, $m2->getStudent_index(0));
    
    $this->assertEqual($m, $s1->getMentor_index(0));
    $this->assertEqual($m2, $s1->getMentor_index(1));
    
    $this->assertEqual($m, $s2->getMentor_index(0));
    $this->assertEqual($m, $s3->getMentor_index(0));
  }   
  

  function test_removeStudent()
  {
    $m = new MentorN("blah");
    $s = new StudentN(99);
    
    $m->addStudent($s);
    $m->removeStudent($s);
    
    $this->assertEqual(0,$s->numberOfMentors());
    $this->assertEqual(0,$m->numberOfStudents());
  }  


  function test_deleteAll()
  {
    $m = new MentorN("blah");
    $m2 = new MentorN("blah2");
    
    $s = new StudentN(99);
    $s2 = new StudentN(98);
    $s3 = new StudentN(97);
    
    $m->addStudent($s);
    $m->addStudent($s2);
    $m->addStudent($s3);

    $m2->addStudent($s2);
    
    $m->delete();

    $this->assertEqual(0,$m->numberOfStudents());
    $this->assertEqual(1,$m2->numberOfStudents());
    
    $this->assertEqual(0,$s->numberOfMentors());
    $this->assertEqual(1,$s2->numberOfMentors());
    $this->assertEqual(0,$s3->numberOfMentors());
  }  
  
}

?>
